addon/actions/action.js

```diff
@@ -32,6 +32,10 @@ export default EmberObject.extend({
   init() {
     this._super(...arguments);
     assert('Model has to be persisted!', !(this.get('instance') && !this.get('model.id')));
+
+    let payload = this.get('payload') || {};
```

Comment:
(restored comment)
I've made this for the case when user will pass null as payload (as in your tests). Default parameters works only for undefined (null is still passed as null).
If you don't want to allow for null, than this could be removed.